### PR TITLE
Split Secure Cell spec into pages

### PR DESCRIPTION
**Pull request by ilammy**
There's way too much text on this page, and I guess I'll be writing some more. Let's split it into more pages, we're not going to ruin a rainforest for that.

Since the spec is WIP at this point, we can more or less disregard URL stability. But still, choose something that's going to stick.

### CODE REVIEW

**Review by vixentael — APPROVED** [review]

File: themis/spec/secure-cell/_index.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+weight: 10
+title:  Secure Cell
+bookCollapseSection: true
+---
+
+# Secure Cell
+
+{{< hint info >}}
+This part of Themis documentation is currently under development.
+It might be incomplet and incorekt.
```
Comment (vixentael) [inline themis/spec/secure-cell/_index.md:11]: :)

File: themis/spec/secure-cell/examples.md [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+
+| Input          | Value |
+| -------------- | ----- |
+| encryption key | `au6aimoa8Pee8wahxi4Aique6eaxai2a` |
+| plaintext      | `encrypted message` |
+| context data   | `additional context` |
+
+Secure Cell in Seal mode produces the following output (encoded in base64):
+
+    AAEBQAwAAAAQAAAAEQAAAM5da3KkReYC7++OPbrI13UycoVi3s01Ji64WQ/KIe+3oF8cgLle19WC+tnaCg==
```
Comment (vixentael) [inline themis/spec/secure-cell/examples.md:25]: does it look fine without being surrounded by ```?

File: themis/spec/secure-cell/examples.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+| algorithm ID    | 0x00   | `00 01 01 40` | `0x40010100` – AES-GCM-256, Soter KDF, PKCS#7 padding |
+| IV length       | 0x04   | `0c 00 00 00` | IV data is 12 bytes long |
+| auth tag length | 0x08   | `10 00 00 00` | auth tag is 16 bytes long |
+| message length  | 0x0C   | `11 00 00 00` | payload is 17 bytes long |
+| IV data         | 0x10   | `ce . . . 3d` | initialisation vector data |
+| auth tag        | 0x1C   | `ba . . . 0f` | authentication tag data |
+
+All the encoded values match the expectations.
+
+## Pssphrase encryption
```
Comment (vixentael) [inline themis/spec/secure-cell/examples.md:79]:
```suggestion
## Passphrase encryption
```

File: themis/spec/secure-cell/modes.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+if you can afford to store the auxiliary data elsewhere.
+
+Context Imprint mode exists for use cases which do not allow for any additional storage.
+This constraint somewhat lowers the security of the cryptosystem
+since there is no space for authentication data or – more importantly – random IV.
+To compensate for this, Secure Cell requires *associated data* to be used in Context Imprint mode.
+
+## Associated data and nonces
+
+Another axis of Secure Cell is how much *associated data* the user provides for encryption.
```
Comment (vixentael) [inline themis/spec/secure-cell/modes.md:64]:
```suggestion
Another axis of Secure Cell is how much *associated data* (aka *context*) the user provides for encryption.
```

File: themis/spec/secure-cell/design.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+# Secure Cell design and properties
+
+First of all, let's discuss the desired properties of the cryptosystem.
+
+  - Strong confidentiality and integrity guarantees.
+
+    Given the typical use cases for Secure Cell,
+    it must prevent direct and indirect disclosure of protected data
+    in addition to preventing inadvertent modifications to it.
```
Comment (vixentael) [inline themis/spec/secure-cell/design.md:14]: You realise that –– when published — these lines don't have `\n` between them but rather form a long sentence? Just saying.

File: themis/spec/secure-cell/examples.md (hunk 3 above)
Comment (ilammy) [reply themis/spec/secure-cell/examples.md]: > Pssphrase

😱

Fixed.

File: themis/spec/secure-cell/modes.md (hunk 4 above)
Comment (ilammy) [reply themis/spec/secure-cell/modes.md]: > *associated data* (aka *context*)

Okay, applied.

File: themis/spec/secure-cell/examples.md (hunk 2 above)
Comment (ilammy) [reply themis/spec/secure-cell/examples.md:25]: > does it look fine without being surrounded by ```?

Indentation is a different way to introduce preformatted blocks of text in Markdown. It makes raw text a bit easier to read, so I opt for it from time to time for non-code blocks. And yes, it renders fine.

File: themis/spec/secure-cell/design.md (hunk 5 above)
Comment (ilammy) [reply themis/spec/secure-cell/design.md:14]: > You realise that –– when published — these lines don't have \n between them but rather form a long sentence?

Yes, it's intentional. It's called semantic line-breaking. This makes editing text easier: moving sentences and clauses around, etc.

If a hard line break is really necessary – which is almost never – there's always `<br>`.

**Review by vixentael — APPROVED** [review]

**Pull request merged**